core/include/detray/propagator/rk_stepper.ipp

```diff
@@ -684,9 +712,6 @@ DETRAY_HOST_DEVICE bool detray::rk_stepper<
                           (cfg.rk_error_tol / math::abs(error_estimate))))),
             static_cast<scalar_type>(4));
 
-        // Only step size reduction is allowed so that we don't overstep
-        assert(step_size_scaling <= 1.f);
```

Comment:
if `cfg.rk_error_tol > error_estimate`, `step_size_scaling` gets bigger than one

You can take a look into the illustration in https://iopscience.iop.org/article/10.1088/1748-0221/4/04/P04001/pdf

![image](https://github.com/acts-project/detray/assets/63090140/0bd3ba0d-9f22-45b3-b70f-94ded3003ffc)

Reply:
I know how it is supposed to work, I just don understand our particular implementation? Are you sure that the code branch where the scale factor is calculated can be reached at all when ```(error_estimate <= cfg.rk_error_tol)```?

Reply:
Maybe I just don't get it...

Reply:
what do you mean? you mean that rk_error_tol > error_estimate can not happen?

Reply:
OH isee.. there is try_rk returns false if that happens

Reply:
Why did we write code like this?

Reply:
Exactly. Our algorithm never reaches that point

Reply:
This is the only reason why the navigation works with a totally unconstrained step

Reply:
Now I think it is a wrong design. Let me refactor things so that we can handle the error tolerance correctly. 
At least the current code is not how adadptive RKN is supposed to work

Reply:
I am just not sure what's faster in the end. Trying to step the maximal distance every time or a fully adaptive RK algorithm? Because if you make it adaptive, you will need to add the step size constraints permanently and the logic becomes more involved...

Reply:
Let me see
I don't know what is going to be faster. At least the current code is looping until it passes the tolerance condition.

We can still prevent overshooting issue by simply limiting the stepsize to the navigation path legth.

Reply:
Yes, that't true, if we get around using the ```contstrained_step```, all the better! It will be interesting to see what's faster. I may that we are taking fewer steps right now, that would be good, but may also be that we simply run the stepsize reduction loop more often to arrive at exactly the same number of steps as everyone else. Not so good

Reply:
But we need a sufficiently inhomogeneous B-field to tell, maybe?

Reply:
Volume material will make error large enough I beleive

Reply:
We only have volume material in the wire chamber though, right?

Reply:
I am testing with the telescope geometry with volume material